### PR TITLE
Add qtrx and zebraiken to Super Staff Bros, add faint messages

### PR DESCRIPTION
**Pull request by cadaeic**
Additional changes: updates to LU and Marty's ranks
Comment: Due to the way chat messages are shown in battles, they always appear before the start of animations. The players can 'foresee' whether their mons will faint this turn if these mons' faint messages involve sending chat messages. This probably needs to be changed.

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -2028,6 +2028,11 @@ exports.Formats = [
 				this.add('-immune', pokemon, '[from] Irrelevant');
 				return false;
 			}
+            // qtrx is immune to Torment.
+            if ((type === 'torment' || type === 'taunt') && toId(pokemon.name) === 'qtrx') {
+                this.add('-immune', pokemon);
+                return false;
+            }
```
Comment (xfix) [inline config/formats.js]: Indentation

File: config/formats.js [hunk 2]
```diff
@@ -2063,6 +2068,19 @@ exports.Formats = [
...
+                if (!pokemon.illusion) {
+				    this.add('-start', pokemon, 'typechange', 'Normal/Psychic');
+                    pokemon.typesData = [
+					{type: 'Normal', suppressed: false,  isAdded: false},
+					{type: 'Psychic', suppressed: false,  isAdded: false}
+				    ];
+                }
+				pokemon.addVolatile('focusenergy');
+                pokemon.addVolatile('telekinesis');
+            }
```
Comment (xfix) [inline config/formats.js]: Okay, what's going on with indentation here. Use tabs, don't use spaces for indentation.

File: config/formats.js [hunk 3]
```diff
@@ -2193,9 +2211,22 @@ exports.Formats = [
...
 			}
-			
 			if (name === 'marty') {
-				this.add('c|%Marty|Prepare yourself.');
+				this.add('c|@Marty|Prepare yourself.');
+			}
+			if (name === 'qtrx') {
+                sentences = ["cutie are ex", "q-trix", "quarters", "cute T-rex", "Qatari", "random letters", "spammy letters", "asgdf"];
+                this.add('c|@qtrx|omg DONT call me \'' + sentences[this.random(8)] + '\' pls respect my name its very special!!1!');
+            }
```
Comment (xfix) [inline config/formats.js]: Please... indentation with tabs, not spaces.

File: config/formats.js [hunk 4]
```diff
@@ -2207,12 +2238,341 @@ exports.Formats = [
...
+				sentences = ['that\'s it ur getting banned', 'meow', '(✖╭╮✖)'];
+				this.add('c|@Bean|' + sentences[this.random(3)]);
+			}
+			if (name === 'beowulf') {
+				this.add('c|@Beowulf|There is no need to be mad');
+			}
+			if (name === 'biggie') {
+                sentences = ['It was all a dream', 'It\'s gotta be the shoes', 'ヽ༼ຈل͜ຈ༽ﾉ RIOT ヽ༼ຈل͜ຈ༽ﾉ'];
+                this.add('c|@BiGGiE|' + sentences[this.random(3)]);
+            }
```
Comment (xfix) [inline config/formats.js]: Indentation here as well.

File: config/formats.js [hunk 5]
```diff
@@ -2207,12 +2238,341 @@ exports.Formats = [
...
+					this.add('-message', '░░░░░▀▀▀▀░░░░░░░░░░░░░░░░░');
+				} else if (pokemon.phraseIndex === 1) {
+					this.add('c|@Genesect|┬┴┬┴┤ʕ•ᴥ├┬┴┬┴ well, if that\'s what you want');
+					this.add('c|@Genesect|┬┴┬┴┤ ʕ•├┬┴┬┴');
+					this.add('c|@Genesect|┬┴┬┴┤  ʕ├┬┴┬┴');
+				} else {
+					sentences = ["The darkside cannot be extinguished, when you fight...", "؍༼ಥ_ಥ༽ጋ lament your dongers ؍༼ಥ_ಥ༽ጋ", "Yᵒᵘ Oᶰˡʸ Lᶤᵛᵉ Oᶰᶜᵉ", "やれやれだぜ", " ୧༼ಠ益ಠ༽୨ MRGLRLRLR ୧༼ಠ益ಠ༽୨"].randomize();
+					this.add('c|@Genesect|' + sentences[0]);
+				}
+            }
```
Comment (xfix) [inline config/formats.js]: And here...

File: config/formats.js [hunk 6]
```diff
@@ -2207,12 +2238,341 @@ exports.Formats = [
...
+			if (pokemon.name === 'hippopotas') {
+				this.add('-message', 'The sandstorm subsided.');
+			}
 		},
 		onAfterMoveSelf: function (source, target, move) {
 			// Make Haunter not immune to Life Orb as a means to balance.
 			if (toId(source.name) === 'haunter') {
 				this.damage(source.maxhp / 10, source, source, this.getItem('lifeorb'));
-			}
+			}
```
Comment (xfix) [inline config/formats.js]: Why those spaces?

File: config/formats.js [hunk 7]
```diff
@@ -2222,10 +2582,23 @@ exports.Formats = [
...
+						battle.sides[s].active[p].addVolatile('telekinesis');
+                        if (this.random(2) === 1) {
+                            this.useMove('trickroom', battle.sides[s].active[p]);    
+                        } else {
+                            this.useMove('wonderroom', battle.sides[s].active[p]);
+                        }
+                    }
+					if ((toId(battle.sides[s].active[p].name) === 'beowulf') && !battle.sides[s].active[p].fainted) {
+                        this.add('c|@Beowulf|/me buzzes loudly!');
+                    }
```
Comment (xfix) [inline config/formats.js]: indentation asdf

File: config/formats.js [hunk 8]
```diff
@@ -2677,7 +3063,27 @@ exports.Formats = [
...
+                        if (!target.fainted) {
+							gibberish = '';
+							for (var j = 0; j < 30; j++) gibberish += String.fromCharCode(97 + this.random(26));
+							this.add('-message', gibberish);
+							this.useMove(hps[this.random(16)], source, target);
+						}
+                    }
+				}
+			}
+            if (move.id === 'gravity' && name === 'relados') {
```
Comment (xfix) [inline config/formats.js]: SUPER INDENTATION BROS.

**Comment by xfix** [comment]
Fix your indentaton, it's totally broken, as File changed tab will show you. Use tabs, not spaces.

**Comment by cadaeic** [comment]
@xfix done and done. No idea why Edge Code screwed up the indentation though.

**Pull request closed without merging**